### PR TITLE
[REVIEW][FEATURE][GAWE-49] - Implement reset password business logic

### PR DESCRIPTION
**Pull request by fauzanramadhani**
#### Description

Implement reset password business logic.

#### Ticket Reference(s)

[GAWE-49](https://bluehabit.atlassian.net/browse/GAWE-49)

##### UI Changes

> Screenshot or video of UI-related changes

##### Coverage

> Test coverage screenshot.

N/A

#### Note(s)

@triandamai @WhyUReRenldI @azkifairuz

[GAWE-49]: https://bluehabit.atlassian.net/browse/GAWE-49?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/RequestResetPasswordResponse.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Unauthorized copying, publishing of this file, via any medium is strictly prohibited
+ * Proprietary and confidential
+ */
+
+package com.bluehabit.eureka.data.authentication.datasource.remote.response
+
+import com.google.gson.annotations.SerializedName
+
+data class RequestResetPasswordResponse(
```
Comment (triandamai) [inline data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/RequestResetPasswordResponse.kt:12]: langsung aja `data:Any` statusCode,message,sama data sudah di cover di BaseResponse

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/ResetPasswordResponse.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Unauthorized copying, publishing of this file, via any medium is strictly prohibited
+ * Proprietary and confidential
+ */
+
+package com.bluehabit.eureka.data.authentication.datasource.remote.response
+
+import com.google.gson.annotations.SerializedName
+
+data class ResetPasswordResponse(
```
Comment (triandamai) [inline data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/ResetPasswordResponse.kt:12]: langsung aja `data:Any` statusCode,message,sama data sudah di cover di BaseResponse

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/LinkConfirmationResponse.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+data class LinkConfirmationResponse(
+    @SerializedName("statusCode")
+    val statusCode: Int,
+    @SerializedName("data")
+    val data: Data,
+    @SerializedName("message")
+    val message: String,
+)
+
+data class Data(
```
[the remark this answers is not in the document]
Comment (fauzanramadhani) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/LinkConfirmationResponse.kt:21]: done!

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/RequestResetPasswordResponse.kt (hunk 1 above)
Comment (fauzanramadhani) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/RequestResetPasswordResponse.kt:12]: done!

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/ResetPasswordResponse.kt (hunk 2 above)
Comment (fauzanramadhani) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/datasource/remote/response/ResetPasswordResponse.kt:12]: done!

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt [hunk 4]
```diff
@@ -30,16 +29,16 @@ class ResetPasswordRepository @Inject constructor(
...
         httpClient.post(AuthApi.RequestResetPassword()) {
             setBody(RequestResetPasswordRequest(email))
         }
     }
 
     suspend fun linkConfirmation(
         token: String
-    ): Response<LinkConfirmationResponse> =
-        when (val result = safeApiCall<LinkConfirmationResponse> {
+    ): Response<BaseResponse<LinkConfirmationResponse>> =
```
Comment (triandamai) [inline data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt:40]: ga perlu `BaseResponse` langsung aja `Response<LinkCOnfirmationResponse>`

Comment (triandamai) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt:40]: karena dari `safeApiCall` suda menghandle

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt [hunk 5]
```diff
@@ -56,7 +55,7 @@ class ResetPasswordRepository @Inject constructor(
 
     suspend fun resetPassword(
         newPassword: String
-    ): Response<ResetPasswordResponse> = safeApiCall {
+    ): Response<BaseResponse<Any>> = safeApiCall {
```
Comment (triandamai) [inline data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt:58]: ga perlu `BaseResponse `langsung aja `Response<LinkCOnfirmationResponse>`

**Review by triandamai — CHANGES_REQUESTED** [review]
cek dolo

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt (hunk 4 above)
Comment (fauzanramadhani) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt:40]: done mas

File: data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt (hunk 5 above)
Comment (fauzanramadhani) [reply data/data-authentication/src/main/java/com/bluehabit/eureka/data/authentication/repositories/ResetPasswordRepository.kt:58]: done mas

**Pull request merged**